### PR TITLE
sbctl: update

### PR DESCRIPTION
**Pull request by dkwo**
- I tested the changes in this PR: yes
- I built this PR locally for my native architecture, (x86_64-glibc)

### CODE REVIEW

**Comment by Duncaen** [comment]
I don't think the hook should run by default.

**Comment by Duncaen** [comment]
Kernels are not necessarily named `vmlinuz`, it could be `vmlinux` depending on the architecture.

**Comment by dkwo** [comment]
I could add a check for the existence of vmlinuz.
(Note that efibootmgr also uses vmlinuz, and secure boot is a uefi feature, afaik.)
If one installs sbctl, don't they want to sign new kernels? or what is your concern?

**Comment by Duncaen** [comment]
You have to first create keys and enroll, some users might want to sign kernel with sbctl and move/rename them instead of modifying the tracked binary. 

aarch64 supports efi and the kernel binaries are uncompressed and named `vmlinux`, the `efibootmgr` hook is not very good for multiple reasons.

**Comment by Duncaen** [comment]
We generally don't test for values of those environment variables and instead check if they are set or not, that makes the code easier and avoids having to define what yes/no/true/false/0/1 is the correct choice.

Not sure why you would only sign `vmlinuz` kernels and ignore `vmlinux`.

**Comment by Duncaen** [comment]
[ -z "$SBCTL_SIGN_KERNEL" ] && exit 0
    [ -e "vmlinuz-${VERSION}" ] && sign "boot/vmlinuz-${VERSION}}"
    [ -e "vmlinux-${VERSION}" ] && sign "boot/vmlinuz-${VERSION}}"
    
/etc/default/sbctl-kernel-hook:

    # SBCTL_SIGN_KERNEL=yes

**Comment by dkwo** [comment]
Done, thanks.

File: srcpkgs/sbctl/patches/keyusage.patch [hunk 1]
```diff
@@ -0,0 +1,12 @@
+diff --git a/keys.go b/keys.go
```
Comment (classabbyamp) [inline srcpkgs/sbctl/patches/keyusage.patch:1]: please add a note or link explaining this patch

Comment (dkwo) [reply srcpkgs/sbctl/patches/keyusage.patch:6]: Done.

**Comment by dkwo** [comment]
added a post-removal hook as well.

**Comment by dkwo** [comment]
can this be merged?

**Comment by dkwo** [comment]
update to 0.12, enable pie (is there a smarter way?), enable checks

**Comment by dkwo** [comment]
ready to merge, imo

**Comment by dkwo** [comment]
patch is merged upstream

**Comment by dkwo** [comment]
updated to 0.13, which includes my patch.

**Comment by Duncaen** [comment]
> make it use /etc/secureboot instead, via a go_ldlfag

Sucks that its in `/usr/share`, but breaking already existing installs and diverting from the documentation doesn't seem right.

**Comment by Duncaen** [comment]
I would probably prefer to not change it now that it has been like this in the repos for 3 years and we don't know what upstream is going to do.

There are 3 scenarios:
- Upstream changes it to what we "patched" in, everythink good.
- Upstream changes it to something else, users have to manually change again.
- Upstream changes it and adds a fallback mechanism, we don't have to do anything.

**Comment by dkwo** [comment]
I removed the kernel hook, as I prefer to sign the unified kernel image. Also simpler in view of moving kerneks out of /boot.
This is now just an update, which i've been using for months with no issues.

**Comment by dkwo** [comment]
0.14 is out, seems to work fine.

**Comment by Calandracas606** [comment]
version 15 is out: https://github.com/Foxboron/sbctl/releases/tag/0.15

**Comment by dkwo** [comment]
Thanks. I've tested the new version, and it works for me, as well as the `migrate` subcommand.
I'm open to suggestions as for the `sbctl.conf` file, which I've taken from upstream.
I do not create `/var/lib/sbctl` using make_dirs, as currently the program would complain otherwise
and refuse to proceed if that folder is already present.

@Duncaen This version solves the issue with /usr/share, as it finally moves it to /var/lib/sbctl.
Let me know if an install message is needed to warn the user.

In general, it may be better to wait a few days until things are more stable, and more people have tested this,
so marking as draft.

**Comment by dkwo** [comment]
@Foxboron Thanks a lot! I removed the conf file, and now the package also creates /var/lib/sbctl as expected.

**Comment by dkwo** [comment]
This has been working fine for me.

**Comment by classabbyamp** [comment]
0.15.4 is out

**Comment by dkwo** [comment]
Updated, still working fine.

**Pull request merged**